addons/html_editor/static/src/main/link/link_selection_plugin.js

```diff
@@ -55,90 +33,43 @@ function isLegitZwnbsp(textNode) {
 
 export class LinkSelectionPlugin extends Plugin {
     static id = "linkSelection";
-    static dependencies = ["selection"];
+    static dependencies = ["selection", "feff"];
     // TODO ABD: refactor to handle Knowledge comments inside this plugin without sharing padLinkWithZwnbsp.
     static shared = ["padLinkWithZwnbsp"];
     resources = {
         /** Handlers */
         selectionchange_handlers: this.resetLinkInSelection.bind(this),
-        clean_handlers: (root) => this.removeFEFFs(root, { preserveSelection: true }),
-        clean_for_save_handlers: this.cleanForSave.bind(this),
-        normalize_handlers: (el) => this.normalize(el || this.editable),
-
+        clean_for_save_handlers: ({ root }) => this.clearLinkInSelectionClass(root),
+        normalize_handlers: () => this.resetLinkInSelection(),
+        feff_providers: this.addFeffsToLinks.bind(this),
         system_classes: ["o_link_in_selection"],
-        intangible_char_for_keyboard_navigation_predicates: (ev, char, lastSkipped) =>
-            // Skip first FEFF, but not the second one (unless shift is pressed).
-            char === "\uFEFF" && (ev.shiftKey || lastSkipped !== "\uFEFF"),
     };
 
-    /**
-     * @param {Element} root
-     */
-    normalize(root) {
-        this.updateFEFFs(root);
-        this.resetLinkInSelection();
-    }
-
-    /**
-     * @param {Element} root
-     */
-    cleanForSave({ root, preserveSelection = false }) {
-        this.removeFEFFs(root, { preserveSelection });
-        this.clearLinkInSelectionClass(root);
-    }
-
-    /**
-     * @param {Element} root
-     */
-    updateFEFFs(root) {
-        this.removeFEFFs(root, { exclude: isLegitZwnbsp });
-
-        for (const link of selectElements(root, "a")) {
-            if (this.isLinkEligibleForZwnbsp(link)) {
-                // Only add the ZWNBSP for simple (possibly styled) text links, and
-                // never in a nav.
-                this.padLinkWithZwnbsp(link);
-            }
-        }
+    addFeffsToLinks(root, cursors) {
```

Comment:
Stable retrocompatibility: put back the old function names that will just call their equivalent in feff plugin.